verifier/src/report_body.rs

```diff
@@ -258,6 +261,38 @@ impl IntoVerificationMetadata for MrEnclave {
     }
 }
 
+/// Metadata for [`MrSigner`] verification
+#[derive(Debug, Clone)]
+pub struct MrSignerVerificationMetadata {
+    mr_signer_key: VerificationResult<VerificationResultMetadata>,
+    product_id: VerificationResult<VerificationResultMetadata>,
+    isv_svn: VerificationResult<VerificationResultMetadata>,
+}
```

Comment:
Using `MrSignerVerificationMetadata` instead of another variant in `VerificationResultMetadata` because of the ownership of these 3 members.  If this was another variant there would be a recursion issue of a `VerificationREsultMetadata` containing a `VerificationResultMetadata`.  This would require something like `Box` to alleviate, but then `Box` brings in alloc.